docs/Developers/Integration/PTOracles.md

```diff
@@ -0,0 +1,106 @@
+---
+hide_table_of_contents: true
+---
+
+# PT Oracles
+
+
+The **Principal Token (PT)** is an ERC20 token bearing the value of a certain amount of asset which can be redeemed at maturity. For example, `100 PT-stETH-25DEC2025` can be redeemed for `100 stETH` on the 25th of December 2025.
+
+In Pendle system, $PT$ can be freely traded from and to $SY$ (EIP-5115 token) ultilizing our AMM. With the built-in TWAP oracle library, the geometric mean price of $PT$ can be derived from our `PendleMarket` contracts fully on-chain.
+
+## About Our Oracles
+
+Pendle's oracle implementation is inspired from the idea of UniswapV3 Oracle (see [here](https://docs.uniswap.org/concepts/protocol/oracle)) with a slight difference on how we define the cumulative rate. In short, our oracle stores the cumulative logarithm exchange rate between $PT$ and $Asset$ (please refer to the definition of assets [here](https://eips.ethereum.org/EIPS/eip-5115)).
+
+### EIP-5115 Asset
+
+Although the AMMs hold their positions and execute trades in $PT$ and $SY$, their underlying logic assumes all their $SY$ balances are converted to assets. As a result, the exchange rate stored in our oracles are between $PT$ and $Asset$, which should make pricing $PT$ more straightforward.
```

Comment:
"the exchange rate **returned by our oracles** are between PT and Asset" (since the exchange rate is not actually stored. What we stored in the contracts are the implied rates, not the exchange rates.